### PR TITLE
Fixed Encypt storage migration issue

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-2236/pillar-wallet-sqlite-encryption

## Description

- Fixed Encryption storage migration 

## Motivation and Context

-

## How Has This Been Tested?

- Tested on Android and iOS device

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/services/encryptedStorage.js [hunk 1]
```diff
@@ -70,6 +78,22 @@ EncryptedStorage.prototype.save = async function (id: string, data: Object, forc
...
+  if (isEmpty(wallet)) return Promise.resolve();
+
+  try {
+    printLog('Encrypted Storage migrating data');
+    // eslint-disable-next-line i18next/no-literal-string
+    await this.save('wallet', { wallet });
+    // eslint-disable-next-line i18next/no-literal-string
+    await storage.remove('wallet');
+  } catch (e) {
+    reportErrorLog('AsyncStorage migration to Encrypted Storage failed', { error: e });
```
Comment (IAmKio) [inline src/services/encryptedStorage.js:92]: If this fails - what happens?

File: ios/pillarwallet/AppDelegate.mm [hunk 2]
```diff
@@ -8,10 +8,34 @@
...
+            SecItemDelete((__bridge CFDictionaryRef)spec);
+        }
+    }
+}
+
 @implementation AppDelegate
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
+  ClearKeychainIfNecessary();
```
Comment (IAmKio) [inline ios/pillarwallet/AppDelegate.mm:37]: can you just explain in what scenario this would be needed?

File: src/actions/appActions.js [hunk 3]
```diff
@@ -190,29 +192,29 @@ export const initAppAndRedirectAction = () => {
...
-          type: TODAY_FAILED_ATTEMPTS,
-          payload: {
-            failedAttempts: {
-              numberOfFailedAttempts,
-              date: new Date(date),
-            },
-          },
-        });
-      }
+      // const currentTime = await getCurrentTime();
```
Comment (IAmKio) [inline src/actions/appActions.js:195]: Do we still actually need this or is it commented out for a reason?

**Review by IAmKio — CHANGES_REQUESTED** [review]

File: ios/pillarwallet/AppDelegate.mm (hunk 2 above)
Comment (lalitkumarrock) [reply ios/pillarwallet/AppDelegate.mm:37]: Actually, encrypted storage data is stored in keychain.  still, data is stored in the keychain if we uninstall the app. used this to remove data when first time open app

File: src/services/encryptedStorage.js (hunk 1 above)
Comment (IAmKio) [reply src/services/encryptedStorage.js:92]: @lalitkumarrock I know sorry but why?

Comment (IAmKio) [reply src/services/encryptedStorage.js:92]: I just need to know what is the fallback scenario here

File: ios/pillarwallet/AppDelegate.mm (hunk 2 above)
Comment (IAmKio) [reply ios/pillarwallet/AppDelegate.mm:37]: Got it thanks

File: src/services/encryptedStorage.js (hunk 1 above)
Comment (lalitkumarrock) [reply src/services/encryptedStorage.js:92]: Here failure chances are very low. if fails then open the onboarding screen.
I pushed one changes. [de04d01](https://github.com/pillarwallet/pillarwallet/pull/3488/commits/de04d0106f81c56b86d736433000c01877986647)

**Review by IAmKio — APPROVED** [review]

**Pull request merged**